File: mobile/android/FUN/app/src/main/java/com/fun/app/data/models/Comment.kt
/**
 * FUN App - Comment Models
 */

package com.fun.app.data.models

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class Comment(
    @Json(name = "_id") val id: String,
    val userId: String,
    val seriesId: String,
    val episodeNum: Int?,
    val text: String,
    val createdAt: String,
    val user: CommentUser?
)

@JsonClass(generateAdapter = true)
data class CommentUser(
    val displayName: String,
    val avatarUrl: String?
)

@JsonClass(generateAdapter = true)
data class CommentsResponse(
    val comments: List<Comment>,
    val pagination: Pagination
)

@JsonClass(generateAdapter = true)
data class PostCommentRequest(
    val seriesId: String,
    val episodeNum: Int?,
    val text: String
)

@JsonClass(generateAdapter = true)
data class PostCommentResponse(
    val comment: Comment
)

File: mobile/ios/FUN/FUN/Models/Comment.swift
/**
 * FUN App - Comment Model
 */

import Foundation

struct Comment: Codable, Identifiable {
    let id: String
    let userId: String
    let seriesId: String
    let episodeNum: Int?
    let text: String
    let createdAt: String
    let user: CommentUser?
    
    enum CodingKeys: String, CodingKey {
        case id = "_id"
        case userId, seriesId, episodeNum, text, createdAt, user
    }
}

struct CommentUser: Codable {
    let displayName: String
    let avatarUrl: String?
}

struct CommentsResponse: Codable {
    let comments: [Comment]
    let pagination: Pagination
}

struct PostCommentRequest: Codable {
    let seriesId: String
    let episodeNum: Int?
    let text: String
}

struct PostCommentResponse: Codable {
    let comment: Comment
}
